packages/zowe-explorer-ftp-extension/src/ZoweExplorerFtpJesApi.ts

```diff
@@ -0,0 +1,249 @@
+                    jobid: jobDetails.jobid,
+                    jobname: jobDetails.jobname,
+                    recfm: "FB",
+                    lrecl: 80,
+                    "byte-count": spoolFileToDownload.byteCount,
+                    // todo is recfm or lrecl available? FB 80 could be wrong
+                    "record-count": 0,
+                    "job-correlator": "", // most of these options don't matter for download
+                    class: "A",
+                    ddname: spoolFileToDownload.ddname,
+                    id: spoolFileToDownload.id,
+                    "records-url": "",
+                    subsystem: "JES2",
+                    stepname: spoolFileToDownload.stepname,
+                    procstep:
+                        spoolFileToDownload.procstep === "N/A" || spoolFileToDownload.procstep == null
+                            ? undefined
+                            : spoolFileToDownload.procstep,
+                };
+                const destinationFile = DownloadJobs.getSpoolDownloadFile(
+                    mockJobFile,
+                    parms.omitJobidDirectory,
+                    parms.outDir
+                );
+                imperative.IO.createDirsSyncFromFilePath(destinationFile);
+                imperative.IO.writeFile(destinationFile, spoolFileToDownload.contents);
+            }
+        }
+        return;
+    }
+
+    public async getSpoolContentById(jobname: string, jobid: string, spoolId: number): Promise<string> {
+        const connection = await this.ftpClient(this.checkedProfile());
+        if (connection) {
+            const options = {
+                fileId: spoolId.toString(),
+                jobName: jobname,
+                jobId: jobid,
+                owner: "*",
+            };
+            const response: Buffer = await JobUtils.getSpoolFileContent(connection, options);
+            if (response) {
+                // eslint-disable-next-line @typescript-eslint/no-unsafe-return
+                return response.toString();
+            }
+        }
+        return "";
+    }
+
+    public getJclForJob(job: zowe.IJob): Promise<string> {
+        throw new Error("Get jcl is not supported.");
+    }
+
+    public submitJcl(jcl: string, internalReaderRecfm?: string, internalReaderLrecl?: string): Promise<zowe.IJob> {
+        throw new Error("Submit jcl is not supported.");
+    }
+
+    public async submitJob(jobDataSet: string): Promise<zowe.IJob> {
+        const result = this.getIJobResponse();
+        const connection = await this.ftpClient(this.checkedProfile());
+        if (connection) {
+            const transferOptions = {
+                transferType: TRANSFER_TYPE_ASCII,
+            };
+            const content = await DataSetUtils.downloadDataSet(connection, jobDataSet, transferOptions);
+            const jcl = content.toString();
+            const response: string = await JobUtils.submitJob(connection, jcl);
```

Comment:
`jobId` is more informative than `response`.

Reply:
done.